### PR TITLE
feat: update base image with debian-base

### PR DESCRIPTION
**Pull request by sozercan**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
- Updates dockerfile with a builder image to build `azurediskplugin` (for building inside the container)
- Updates base image with `us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.1.0` as minimal base (https://github.com/kubernetes/kubernetes/tree/master/build/debian-base)
- Uses `docker buildx` to build cross-platform images

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
feat: replace base image with debian-base-amd64:v2.1.0
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -94,33 +94,34 @@ e2e-teardown:
...
+	CGO_ENABLED=0 GOOS=linux go build -a -ldflags "${LDFLAGS}" -o _output/azurediskplugin ./pkg/azurediskplugin
 
 .PHONY: azuredisk-windows
 azuredisk-windows:
 	if [ ! -d ./vendor ]; then dep ensure -vendor-only; fi
-	CGO_ENABLED=0 GOOS=windows go build -a -ldflags ${LDFLAGS} -o _output/azurediskplugin.exe ./pkg/azurediskplugin
+	CGO_ENABLED=0 GOOS=windows go build -a -ldflags "${LDFLAGS}" -o _output/azurediskplugin.exe ./pkg/azurediskplugin
 
 .PHONY: container
-container: azuredisk
```
Comment (andyzhangx) [inline Makefile:105]: make keep this, otherwise `make container` won't build linux binary

Comment (sozercan) [reply Makefile:105]: Moved that inside the container now, so should build container regardless of local dev env

File: Makefile [hunk 2]
```diff
@@ -102,25 +102,24 @@ azuredisk-windows:
...
-	az acr build --registry $(REGISTRY_NAME) -t $(IMAGE_TAG)-windows-1809-amd64 -f ./pkg/azurediskplugin/Windows.Dockerfile --platform windows .
+	docker buildx build --no-cache --build-arg LDFLAGS=${LDFLAGS} -t $(IMAGE_TAG)-linux-amd64 -f ./pkg/azurediskplugin/Dockerfile --platform="linux/amd64" --push .
+	docker buildx build --no-cache --build-arg LDFLAGS=${LDFLAGS} -t $(IMAGE_TAG)-windows-1809-amd64 -f ./pkg/azurediskplugin/Windows.Dockerfile --platform="windows/amd64" --push .
 	docker manifest create $(IMAGE_TAG) $(IMAGE_TAG)-linux-amd64 $(IMAGE_TAG)-windows-1809-amd64
 	docker manifest inspect $(IMAGE_TAG)
 else
 ifdef TEST_WINDOWS
-	make azuredisk-windows
-	docker build --no-cache -t $(IMAGE_TAG) -f ./pkg/azurediskplugin/Windows.Dockerfile .
+	docker buildx build --no-cache --build-arg LDFLAGS="${LDFLAGS}" -t $(IMAGE_TAG)-windows-1809-amd64 -f ./pkg/azurediskplugin/Windows.Dockerfile --platform="windows/amd64" --output "type=docker,push=false" .
```
Comment (jchauncey) [inline Makefile:120]: I had to remove the quotes from LDFLAGS to get this to work

**Pull request merged**